### PR TITLE
Input layer fixes

### PR DESCRIPTION
**Pull request by Mumsfilibaba**
## Purpose
 * Hopefully fixes the bug where the input-layer is stuck in debug
* Adds the IsKeyJustPressed. This function checks if a key were pressed this frame

## Changes
* Added a new struct for KeyState containing the state of the key and if it were pressed this frame.
* Removed preprocessor statement in production that disabled some functions depending on the input layer. This should probably be there for all builds since the console is available at all times. 

## Testing
How have one tested the feature to ensure it works?
- [x] Tested in Sandbox
- [ ] Tested in Multiplayer with 2 clients
- [ ] Tested in Multiplayer with more than 2 clients
- [ ] Tested in Benchmark

### CODE REVIEW

File: CrazyCanvas/Source/GUI/EscapeMenuGUI.cpp [hunk 1]
```diff
@@ -127,31 +121,21 @@ void EscapeMenuGUI::OnButtonBackClick(Noesis::BaseComponent* pSender, const Noes
 
 void EscapeMenuGUI::OnButtonResumeClick(Noesis::BaseComponent* pSender, const Noesis::RoutedEventArgs& args)
 {
-	UNREFERENCED_VARIABLE(pSender);
```
Comment (norda97) [inline CrazyCanvas/Source/GUI/EscapeMenuGUI.cpp:130]: Did you mean to remove UNREFS ?

File: CrazyCanvas/Source/GUI/EscapeMenuGUI.cpp [hunk 2]
```diff
@@ -127,31 +121,21 @@ void EscapeMenuGUI::OnButtonBackClick(Noesis::BaseComponent* pSender, const Noes
...
 	Noesis::FrameworkElement* pElement = m_ContextStack.top();
 	pElement->SetVisibility(Noesis::Visibility_Collapsed);
 	m_ContextStack.pop();
-	Input::PopInputMode();
+	Input::PopInputLayer();
 }
 
 void EscapeMenuGUI::OnButtonSettingsClick(Noesis::BaseComponent* pSender, const Noesis::RoutedEventArgs& args)
 {
-	UNREFERENCED_VARIABLE(pSender);
```
Comment (norda97) [inline CrazyCanvas/Source/GUI/EscapeMenuGUI.cpp:148]: Did you mean to remove UNREFS ?

File: CrazyCanvas/Source/GUI/EscapeMenuGUI.cpp [hunk 3]
```diff
@@ -162,13 +146,8 @@ void EscapeMenuGUI::OnButtonSettingsClick(Noesis::BaseComponent* pSender, const
 
 void EscapeMenuGUI::OnButtonLeaveClick(Noesis::BaseComponent* pSender, const Noesis::RoutedEventArgs& args)
 {
-	UNREFERENCED_VARIABLE(pSender);
```
Comment (norda97) [inline CrazyCanvas/Source/GUI/EscapeMenuGUI.cpp:165]: Did you mean to remove UNREFS ?

File: CrazyCanvas/Source/GUI/EscapeMenuGUI.cpp [hunk 4]
```diff
@@ -180,26 +159,21 @@ void EscapeMenuGUI::OnButtonLeaveClick(Noesis::BaseComponent* pSender, const Noe
...
 	StateManager::GetInstance()->EnqueueStateTransition(pMainMenuState, STATE_TRANSITION::POP_AND_PUSH);
 
-	Input::PopInputMode();
+	Input::PopInputLayer();
 }
 
 void EscapeMenuGUI::OnButtonExitClick(Noesis::BaseComponent* pSender, const Noesis::RoutedEventArgs& args)
 {
-	UNREFERENCED_VARIABLE(pSender);
-	UNREFERENCED_VARIABLE(args);
```
Comment (norda97) [inline CrazyCanvas/Source/GUI/EscapeMenuGUI.cpp:189]: Did you mean to remove UNREFS ?

File: CrazyCanvas/Source/GUI/EscapeMenuGUI.cpp [hunk 5]
```diff
@@ -277,13 +248,8 @@ void EscapeMenuGUI::OnButtonCancelSettingsClick(Noesis::BaseComponent* pSender,
 
 void EscapeMenuGUI::OnButtonChangeControlsClick(Noesis::BaseComponent* pSender, const Noesis::RoutedEventArgs& args)
 {
-	UNREFERENCED_VARIABLE(pSender);
```
Comment (norda97) [inline CrazyCanvas/Source/GUI/EscapeMenuGUI.cpp:280]: Did you mean to remove UNREFS ?

File: Assets/Shaders/PostProcess/TAA_accumulation.frag [hunk 6]
```diff
@@ -292,7 +291,7 @@ void main()
 	currentSample = Resolve(mix(currentSample, historySample, weight));
 
 	// Maniplulate history
-#if 1
+#if 0
 	currentSample.a = currentSample.a * TAA_HISTORY_DECAY;
 #else
 	currentSample.a = weight;
```
Comment (skumbananaaa) [inline Assets/Shaders/PostProcess/TAA_accumulation.frag:297]: Är detta rätt?

**Review by skumbananaaa — APPROVED** [review]
Smexo

**Review by norda97 — APPROVED** [review]
Good job, take look at those UNREFRENCE_VARIABLE you removed in GUI files

**Pull request merged**